mgmt2/config/FileManager.cc

```diff
@@ -91,12 +91,11 @@ FileManager::FileManager()
   this->registerCallback(&handle_file_reload);
 
   // Register the files registry jsonrpc endpoint
-  rpc::add_method_handler(
-    "filemanager.get_files_registry",
-    [this](std::string_view const &id, const YAML::Node &req) -> ts::Rv<YAML::Node> {
-      return get_files_registry_rpc_endpoint(id, req);
-    },
-    &rpc::core_ats_rpc_service_provider_handle);
+  rpc::add_method_handler("filemanager.get_files_registry",
+                          [this](std::string_view const &id, const YAML::Node &req) -> ts::Rv<YAML::Node> {
+                            return get_files_registry_rpc_endpoint(id, req);
+                          },
+                          &rpc::core_ats_rpc_service_provider_handle, {{rpc::NON_RESTRICTED_API}});
```

Comment:
Seems dangerous, capturing the **this** pointer, tricky to keep it from not becoming stale due to object destruction:
```
wkaras ~/REPOS/MISC/LAMBDA_THIS
$ cat x.cc
#include <iostream>

struct A;

A *ga;

struct A
{
    int i = 5;

    auto g() { return [this]() -> A & { return *this; }; }

    auto g2()
    {
        A a{*this};

        delete ga;

        return [a]() -> int { return a.i; };
    }
};

auto h()
{
    return ga->g();
}

int main()
{
    ga = new A;

    A *pa = &(h()());

    std::cout << ga->g2()() << '\n';

    if (pa == ga)
    {
        std::cout << "SAME\n";
    }
    else
    {
        std::cout << "NOT SAME\n";
    }

    return 0;
}
wkaras ~/REPOS/MISC/LAMBDA_THIS
$ gcc x.cc -lstdc++
wkaras ~/REPOS/MISC/LAMBDA_THIS
$ ./a.out
5
SAME
wkaras ~/REPOS/MISC/LAMBDA_THIS
$ 
```
The g2() function shows safe way to capture an object whose class the function is a member of.

Reply:
`FileManager` is now a singleton class, by the time it's gone it shouldn't dangle.

Reply:
:open_mouth:

Reply:
I don't want to add unrelated changes to this PR -> https://github.com/apache/trafficserver/pull/8888 @ywkaras